mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
Looking at io.ReadFull, we might do better by providing our out readFull. This can happen here or in another PR.

We still need a check for io.EOF in the read loops for vector and matrix elements. Replace the err with io.ErrUnexpectedEOF in that case (ReadFull returns EOF is the read was zero length - so we want to return that an incomplete matrix was read).

After that (with or without our own readFull) LGTM.